src/Parser/Parser.hs

```diff
@@ -354,12 +370,12 @@ xtype :: Parser Xtype
 xtype = (try $ do
   x1 <- xtype'
   reservedOp "&"
-  enum' <- enum
+  enum' <- enumName
   case x1 of
     (X b xs) -> return (X b (S.union xs enum'))
     a -> return a)
   <|>
-      xtype'
+      try (xtype' <* (notFollowedBy (string "&")))
```

Comment:
This is a seemingly spurious try, but if we consume input here, the error message is worse. We get "unexpected "&" instead of this:

![parseError](https://user-images.githubusercontent.com/6590681/97792577-3fda0400-1b9d-11eb-855e-467758091769.png)

Also, note that it reports "unexpected f" and the source of the error on line 8. This is the same problem that we have in other parse errors.

Reply:
I did not bother putting too much more effort into it since the message is quite informative and in the future, we will likely catch it as a type error while building the type environment from type assignments.

Reply:
That sounds good to me for the moment.